source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -207,3 +225,511 @@ void AMContinuousScanActionController::fillDataMaps(AMAgnosticDataAPIDataAvailab
+	AMDetectorContinuousMotionRangeData scalerRangeData = scalerChannelDetectors_.first()->retrieveContinuousMotionRangeData(scalerBaseData, expectedDurationScaledToBaseTimeScale_, 20);
+	if(scalerRangeData.isValid()){
+		scalerInitiateMovementIndex = scalerRangeData.motionStartIndex();
+	}
+
+
+	// Not general yet ... need to pass in which channel we should look in for scaler movement data
+	encoderUpVector_ = scalerChannelRebaseVectors_.value("EncoderUp");
+	encoderDownVector_ = scalerChannelRebaseVectors_.value("EncoderDown");
+
+	if( (encoderUpVector_.count()-scalerInitiateMovementIndex) < expectedDurationScaledToBaseTimeScale_){
+		qDebug() << "Scaler start movment index was out of bounds at " << scalerInitiateMovementIndex;
+		scalerInitiateMovementIndex += ((encoderUpVector_.count()-scalerInitiateMovementIndex)-expectedDurationScaledToBaseTimeScale_);
+		scalerRangeData.setMotionStartIndex(scalerInitiateMovementIndex);
+		qDebug() << "Changed to " << scalerInitiateMovementIndex;
+	}
+	if(!amptekDetectors_.isEmpty()){
+		AMDSClientDataRequest *oneAmptekRequest = clientDataRequestMap_.value(amptekDetectors_.first()->amdsBufferName());
+		if( (oneAmptekRequest->data().count() - amptekRangeData.motionStartIndex()) < expectedDurationScaledToBaseTimeScale_){
+			qDebug() << "Amptek start movment index was out of bounds at " << amptekRangeData.motionStartIndex();
+			amptekRangeData.setMotionStartIndex( ((oneAmptekRequest->data().count() - amptekRangeData.motionStartIndex()) - expectedDurationScaledToBaseTimeScale_) + amptekRangeData.motionStartIndex() );
+			qDebug() << "Changed to " << amptekRangeData.motionStartIndex();
+		}
+	}
+
+	foreach(CLSAMDSScalerChannelDetector *scalerChannelDetector, scalerChannelDetectors_)
+		detectorStartMotionIndexMap_.insert(scalerChannelDetector->name(), scalerRangeData.motionStartIndex());
+	if(!amptekDetectors_.isEmpty()){
+		foreach(CLSAmptekSDD123DetectorNew* amptekDetector, amptekDetectors_){
+			detectorStartMotionIndexMap_.insert(amptekDetector->name(), amptekRangeData.motionStartIndex());
+		}
+	}
+
+	qDebug() << "Amptek: " << amptekRangeData.motionStartIndex() << amptekRangeData.motionEndIndex() << amptekRangeData.listIndex();
+	qDebug() << "Scaler: " << scalerRangeData.motionStartIndex() << scalerRangeData.motionEndIndex() << scalerRangeData.listIndex();
+
+	scalerInitiateMovementIndex_ = detectorStartMotionIndexMap_.value(scalerChannelDetectors_.first()->name());
+
+	return true;
+}
+
+bool AMContinuousScanActionController::generateAxisFeedbackValues()
+{
+	return false;
+}
+
+bool AMContinuousScanActionController::generateInterpolatedParameters()
+{
+	double startAxisValue = axisFeedbackValues_.first();
+	double endAxisVaue = axisFeedbackValues_.last();
+	if(startAxisValue > endAxisVaue)
+		qSwap(startAxisValue, endAxisVaue);
+
+	qDebug() << "Start energy was " << startAxisValue << " end energy was " << endAxisVaue;
+	startAxisValue = ceilf(startAxisValue * (1/resolutionStep_)) / (1/resolutionStep_);
```

Comment:
For readability, it might be better to make your `1/resolutionStep_` a variable or maybe do `startAxisValue = ceilf(startAxisValue * (1/resolutionStep_)) * resolutionStep_;`